promql/engine.go

```diff
@@ -1039,12 +1039,26 @@ type groupedAggregation struct {
 }
 
 // aggregation evaluates an aggregation operation on a vector.
-func (ev *evaluator) aggregation(op itemType, grouping model.LabelNames, keepExtra bool, vec vector) vector {
+func (ev *evaluator) aggregation(op itemType, grouping model.LabelNames, without bool, keepExtra bool, vec vector) vector {
 
 	result := map[uint64]*groupedAggregation{}
 
 	for _, sample := range vec {
-		groupingKey := model.SignatureForLabels(sample.Metric.Metric, grouping...)
+		var withoutMetric metric.Metric
+		if without {
+			withoutMetric = *sample.Metric.Copy()
```

Comment:
The problem is we need this version in two places, so I didn't want to duplicate the code.

Dropped the line.